### PR TITLE
Cleanup error management with ts-invariant

### PR DESCRIPTION
**Pull request by antho1404**
### Description

Cleanup the codebase of the hooks:
- Remove useless checks (because we now use the graphql request
- Centralize error messages in an `error.ts` file
- Use `ts-invariant` to simplify the codebase and replace all the `if() throw new Error()`

### CODE REVIEW

File: packages/hooks/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 #### Changed
 
+- Use of ts-invariant to assert conditions in the codebase + remove useless checks [#](https://github.com/liteflow-labs/libraries/pull/)
```
Comment (antho1404) [inline packages/hooks/CHANGELOG.md]:
```suggestion
- Use of ts-invariant to assert conditions in the codebase + remove useless checks [#64](https://github.com/liteflow-labs/libraries/pull/64)
```

**Comment by achampagnedev** [comment]
Does it make more sense to rename `ErrorCodes` into `ErrorMessages` ?

**Comment by antho1404** [comment]
Good catch, all renamed ;)

**Review by achampagnedev — APPROVED** [review]

**Comment by achampagnedev** [comment]
Turborepo crash issue but code is ok

**Comment by antho1404** [comment]
Had some issue while generating the package-lock, this has been fixed

**Pull request merged**